application/src/main/java/org/thingsboard/server/service/queue/strategy/TbMsgQueueHandlerStrategy.java

```diff
@@ -13,17 +13,14 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.thingsboard.rule.engine.api;
+package org.thingsboard.server.service.queue.strategy;
 
-import com.google.common.util.concurrent.FutureCallback;
 import org.thingsboard.server.common.msg.TbMsg;
-import org.thingsboard.server.common.msg.TbMsgPack;
-import org.thingsboard.server.common.msg.TbMsgSubscriptionParams;
 
-public interface TbMsgQueueConsumer {
+import java.util.Map;
+import java.util.UUID;
 
-    void subscribe(TbMsgSubscriptionParams subscriptionParams, FutureCallback<TbMsgPack> callback);
-
-    void ack(TbMsg tbMsg);
+public interface TbMsgQueueHandlerStrategy {
 
+    void handleFailureMsgs(Map<UUID, TbMsg> msgMap);
```

Comment:
This should be refactored to:
TbMsgQueuePack handleFailureMsgs(TbMsgQueuePack pack)